server/src/main/java/org/opensearch/identity/UserIdentity.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.identity;
+
+import java.util.List;
+
+/**
+ * An authenticated user in the system.
+ */
+public class UserIdentity implements Identity {
+
+    /** No identity information or identity information is unavaliable */
+    public static final AnonymousUser ANONYMOUS = new AnonymousUser();
+
+    /** A persistant identifier that is unqiue to the user */
+    private String id;
+
+    public UserIdentity(final String id) {
+        this.id = id;
+    }
+
+    /** A durable and reusable identifier of this user */
+    public String getId() {
+        return this.id;
+    }
+
+    /** No identity information or identity information is unavaliable */
+    public static final class AnonymousUser extends UserIdentity {
+        static final String ID = "Anonymous";
```

Comment:
I think an empty string may be better to use for Anonymous.  Also in case we want to use a format like UUID for this, we'd not want to create a constant that conflicts with it. 

Trivia: did you know there's a hex color code for Anonymous?[ `#817A6E`](https://encycolorpedia.com/817a6e).   `UUID.fromBytes()` with repetitions of those byte values would be a fun constant.

Reply:
I like your UUID suggestion, great easter egg, as it would limit collisions by accident with a pseudo-reserved identity type